### PR TITLE
Optimize data access patterns

### PR DESCRIPTION
**Pull request by markus-gsf-seidl**
Conversion of #53 .

### CODE REVIEW

File: docs/catalog/cloud/use-optimal-software-patterns-for-data.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+version: 1.0
+submitted_by: markus-ntt-seidl
+published_date: TBD
+category: cloud
+tags: 
+ - compute
```
Comment (dubrie) [inline docs/catalog/cloud/use-optimal-software-patterns-for-data.md:7]: please add cloud to the tag section

File: docs/catalog/cloud/use-optimal-software-patterns-for-data.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Use optimal software patterns and architectures for data access and storage
+
+## Description
+
+Application generally need to work with data and every application has specific usage patterns which in turn define in which patterns data is accessed. Optimizing the storage for these type of patterns or optimizing the access to the storage to an optimal pattern can lead to lower carbon usage.
+
+
+## Solution
+
+Monitor and analyze the application to identify if the data access or storage technology is optimal for the use cases of the application. Optimize the access or switch to a more optimal storage technology.
```
Comment (dubrie) [inline docs/catalog/cloud/use-optimal-software-patterns-for-data.md:21]: I'll leave this discussion for the SME review but I'm wondering if the level of this pattern is too high of a level and these potential solutions should be their own patterns.

**Comment by dubrie** [comment]
assigning to @greenhsu123 for SME Review

File: docs/catalog/cloud/use-optimal-software-patterns-for-data.md (hunk 1 above)
Comment (greenhsu123) [inline docs/catalog/cloud/use-optimal-software-patterns-for-data.md:7]: I would argue this is more of a storage pattern than a compute one. Thoughts?

File: docs/catalog/cloud/use-optimal-software-patterns-for-data.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Concerning the SCI equation, using optimal access patterns and storage technology, will impact two parts:
+
+- `E`: Can reduce the embodied emissions if the application after the optimization uses less hardware overall
+- `I`: Optimizing the application will lead to less electricity consumed over time
+
+## Assumptions
+
+- Optimization needs proficiency in the tools and time to be implemented.
+- Changing the storage technology can lead to a large rewrite of the application.
```
Comment (greenhsu123) [inline docs/catalog/cloud/use-optimal-software-patterns-for-data.md:40]: I would argue this is more of a consideration than an assumption.

File: docs/catalog/cloud/use-optimal-software-patterns-for-data.md (hunk 2 above)
Comment (greenhsu123) [reply docs/catalog/cloud/use-optimal-software-patterns-for-data.md:21]: Agreeing with Bill that I think the below should be split out into separate patterns.

Comment (markus-ntt-seidl) [reply docs/catalog/cloud/use-optimal-software-patterns-for-data.md:21]: So a 
"Use optimal software patterns and architectures for data **access**"
and a 
"Use optimal software patterns and architectures for data **storage**"
?

Comment (greenhsu123) [reply docs/catalog/cloud/use-optimal-software-patterns-for-data.md:22]: Not quite! What you have shared above is still one intent mapping to multiple solutions. I think what we are looking for here is a 1:1 mapping. A starting point can be, why is it more sustainable to use in-memory/key-value-like data for session storage?

**Comment by franziska-warncke** [comment]
@markus-gsf-seidl What is blocking here? Should we discuss this in the next meeting?

**Pull request closed without merging**